### DISCUSSION TITLE
How to write file and specify content-type with ObjectStorage

### DISCUSSION THREAD

**Discussion by pedro-cf** [apache/airflow#39648] [Q&A] [answered]
Grettings,

Using [ObjectStorage](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/objectstorage.html#) is it possible to write a file to an Azure Blob Storage container or s3 bucket and specify the content-type?

I was copying `.tif` files from the file system to  an azure lob storage container and the content-type is being set as `application/octet-stream` when it should be `image/tiff`

```python
ObjectStoragePath("file:///opt/airflow/input.tif").copy(dst=base / "copy.tif")
```

In the image bellow `download.tif` and `input.tif` are manually uploaded using Azure Storage Explorer and the remaining files are uploaded using Airflow ObjectStorage:
![image](https://github.com/apache/airflow/assets/4175430/19374439-c2a5-4645-89b7-737e43b8ecf0)

**Comment by pedro-cf** [comment]
Greetings @bolkedebruin perhaps you know the answer to this?

**Comment by bolkedebruin** [comment]
I wonder how you would use this, but a quick search gives:

Azure
https://github.com/fsspec/adlfs/blob/5c24b2e886fc8e068a313819ce3db9b7077c27e3/docs/index.md

For S3 it is unclear if it is supported 
https://github.com/search?q=repo%3Afsspec%2Fs3fs%20contenttype&type=code

**Reply by pedro-cf** [reply]
> Azure https://github.com/fsspec/adlfs/blob/5c24b2e886fc8e068a313819ce3db9b7077c27e3/docs/index.md

Was looking for seamless solution that could work with all object storage, it's ok if it's not possible.

**Reply by pedro-cf** [reply]
> I wonder how you would use this, but a quick search gives:

Not sure what you're asking, but all I want is for the items I upload to object storage to have the proper content-type assigned.

**Comment by pedro-cf** [comment, marked as answer]
https://github.com/apache/airflow/issues/39722
